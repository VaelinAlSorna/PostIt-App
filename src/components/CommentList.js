import React from 'react';
import CommentListItem from './CommentListItem';

const CommentList = (props) => (
         <ol>
            {
                props.comments.map((comment) => <CommentListItem 
                                                userId={props.userId}
                                                onClickDeleteComment={props.onClickDeleteComment} 
                                                key={comment._id} 
                                                {...comment}/>
                                            )
            }
        </ol>
)

export default CommentList;