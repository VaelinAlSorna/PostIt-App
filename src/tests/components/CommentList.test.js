import React from 'react';
import { shallow } from 'enzyme';
import CommentList from '../../components/CommentList';
import articles from '../fixtures/articlesFixtures';

test('should render CommentList correctly', () => {
    const onClickDeleteComment = jest.fn();
    const wrapper = shallow(
                            <CommentList 
                                onClickDeleteComment={onClickDeleteComment} 
                                comments={articles[0].comments} 
                            />
                            );
    expect(wrapper).toMatchSnapshot();
});

