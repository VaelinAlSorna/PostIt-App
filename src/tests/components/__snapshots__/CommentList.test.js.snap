// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render CommentList correctly 1`] = `
<ol>
  <CommentListItem
    _id="1a"
    comment="Something to test?"
    createdAt="1.1.1970"
    key="1a"
    onClickDeleteComment={[Function]}
  />
</ol>
`;
